code/FulpstationCode/fulp_cult/fulp_cult_swords.dm

```diff
@@ -0,0 +1,776 @@
+		return
+
+	var/turf/T = get_turf(A)
+
+	if(!locate(/obj/effect/blessing, T))
+		to_chat(user, "<span class='warning'>The taint of this cursed weapon is too grave to exorcise it off of consecrated grounds.</span>")
+		return
+
+	if(CB.empowered && !CB.possessor)
+
+		if(!cultsword_purge_doafter(CB, user))
+			return
+
+		CB.empowered = FALSE
+		user.visible_message("<span class='notice'>[user] has dissipated [CB]'s unholy energies!</span>")
+		return
+
+	if(CB.possessor)
+
+		if(!cultsword_purge_doafter(CB, user))
+			return
+
+		for(var/mob/M in CB.contents)
+			if(M.mind)
+				M.forceMove(T)
+				new /obj/effect/temp_visual/cult/sparks(M)
+			else(qdel(M))
+
+		qdel(CB)
+		new /obj/effect/decal/cleanable/ash/large(T)
+		playsound(get_turf(loc), 'sound/magic/demon_dies.ogg', 100, TRUE)
+		user.visible_message("<span class='notice'>A dark spirit is exorcised from [CB] as it crumbles to dust!</span>")
+
+
+/obj/item/storage/book/bible/proc/cultsword_purge_doafter(obj/item/melee/cultblade/CB, mob/user)
+
+	var/obj/effect/temp_visual/bless/B
+	to_chat(user, "<span class='notice'>You begin to exorcise [CB].</span>")
+	B = new /obj/effect/temp_visual/bless(CB.loc)
+	playsound(src,pick('sound/hallucinations/veryfar_noise.ogg','sound/hallucinations/growl1.ogg','sound/hallucinations/growl2.ogg','sound/hallucinations/growl3.ogg'),40,TRUE)
+
+	if(!do_after(user, CULT_SWORD_EXORCISE_DELAY, target = CB))
+		qdel(B)
+		return FALSE
+
+	qdel(B)
+	return TRUE
+
+/obj/effect/rune/convert/proc/empower_sword(mob/sacrificial)
+	var/turf/T = get_turf(src)
+	var/obj/item/melee/cultblade/C
+	for(C in T)
+		if(!C.empowered) //We need to be empowered by any sacrifice.
+			C.empowered = TRUE
+			new /obj/effect/temp_visual/cult/sparks(C.loc)
+			new /obj/effect/temp_visual/cult/sparks(loc)
+			Beam(C,icon_state="blood",time=CULT_OFFERING_SWORD_EMPOWER_TIME) //cool sfx
+			playsound(get_turf(loc), 'sound/magic/exit_blood.ogg', 50, TRUE)
+			sacrificial.visible_message("<span class='cultbold'>[C] flares alight with malevolent power!</span>")
+			break //Empower only one.
+
+
+/obj/item/melee/cultblade/equipped(mob/user, slot, initial = FALSE)
+	. = ..()
+	if(!iscultist(user))
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/H = user
+	switch(possessed) //Making this a switch in the event of future on-equip properties
+		if(CULT_HELLFIRE_SWORD)
+			ADD_TRAIT(H, TRAIT_NOFIRE, "hell_blade") //We gain immunity to combustion while we have the sword.
+			H.fire_stacks = 0 //Zero out any existing fire stacks.
+
+
+/obj/item/melee/cultblade/dropped(mob/user)
+	. = ..()
+	if(!iscultist(user))
+		return
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/H = user
+	switch(possessed) //Making this a switch in the event of future on-equip properties
+		if(CULT_HELLFIRE_SWORD)
+			REMOVE_TRAIT(H, TRAIT_NOFIRE, "hell_blade") //We lose immunity to combustion while we don't have the sword.
+			H.fire_stacks = 0 //Zero out any existing fire stacks.

```

Comment:
if youre removing the trait why are you removing stacks? this seems like extra overhead